### DISCUSSION TITLE
Use operator (==, !=, >, >=, <, <=, +, -, *, / ......) for generic-typed variable in a generic type

### DISCUSSION THREAD

**Discussion by vsfeedback** [dotnet/csharplang#3291] [Q&A] [answered]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/idea/878629/use-operator-for-generic-typed-variable-in-a-gener.html)._

---
<p>I have a code snippet where I want to use operator == for the  variable of type  &quot;T where T: struct, Enum&quot;. Also I have benchmarked the performance over using and not using operator == for the generic case and the non-generic case. <br>The question is: Can I use &quot; a ==b&quot; in the method &quot;T_Equals&quot; (in my code snippet)  in the future of C# .NET ? Is it possible ? I want the performance and comfortability. <br><br>=&gt; See the image for the benchmark result.<br></p> <pre>using System;
using System.Diagnostics;

enum E
{
	A = 123, B = 456
}

class Program
{
	static void Main()
	{
		ulong count = 99_999_999;

// T_Equals
		var s = Stopwatch.StartNew();
		for (ulong i = 0; i &lt; count; ++i) T_Equals(E.A, E.B);
		s.Stop();
		Console.WriteLine($&quot;T_Equals: Time= {s.ElapsedMilliseconds} ms, {s.ElapsedTicks} tick&quot;);

// E_Equals
		s = Stopwatch.StartNew();
		for (ulong i = 0; i &lt; count; ++i) E_Equals(E.A, E.B);
		s.Stop();
		Console.WriteLine($&quot;E_Equals: Time= {s.ElapsedMilliseconds} ms, {s.ElapsedTicks} tick&quot;);

// E_operator
		s = Stopwatch.StartNew();
		for (ulong i = 0; i &lt; count; ++i) E_operator(E.A, E.B);
		s.Stop();
		Console.WriteLine($&quot;E_operator: Time= {s.ElapsedMilliseconds} ms, {s.ElapsedTicks} tick&quot;);
	}

static void T_Equals&lt;T&gt;(T a, T b) where T : struct, Enum
	{
		a.Equals(b);
	}

static void E_Equals(E a, E b)
	{
		a.Equals(b);
	}

static void E_operator(E a, E b)
	{
		bool x = a == b;
	}
}
</pre><p><a href="https://developercommunityapi.westus.cloudapp.azure.com/storage/attachments/112849-benchmark-result.png"> Image:112849-benchmark-result.png</a></p>

---
#### Visual Studio Feedback System on 1/8/2020, 11:41 PM: 

Thank you for taking the time to provide your suggestion.  We will do some preliminary checks to make sure we can proceed further.  We'll provide an update once the issue has been triaged by the product team.

**Comment by HyVong007** [comment]
Thank for listening to me !
Hope that new feature will be available soon !

**Comment by CyrusNajmabadi** [comment]
Hey @HyVong007 Can you move this to `dotnet/csharplang`.  That's the right place to make language feature requests.  Thanks! :)

**Comment by huoyaoyuan** [comment]
@HyVong007 FYI:
The feature has been already proposed multiple times. It requires more serious design and runtime change, so it's not trivial and delayed (seems to be C# 10). See dotnet/csharplang#108 .

**Comment by HyVong007** [comment]
> @HyVong007 FYI:
> The feature has been already proposed multiple times. It requires more serious design and runtime change, so it's not trivial and delayed (seems to be C# 10). See [dotnet/csharplang#108](https://github.com/dotnet/csharplang/issues/108) .

 Sad to hear it's not possible now. May be 10 years later it will be possible !

**Comment by gafter** [comment, marked as answer]
This is being considered for C# 10.0 as part of https://github.com/dotnet/csharplang/issues/110.
